uk.co.compucorp.civicrm.hrcomments/CRM/HRComments/API/Query/CommentSelect.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+use Civi\API\SelectQuery;
+use CRM_HRComments_BAO_Comment as Comment;
+
+/**
+ * This class is basically a wrapper around Civi\API\SelectQuery.
+ */
+class CRM_HRComments_API_Query_CommentSelect {
+
+  /**
+   * @var array
+   *   An array of params passed to an API endpoint
+   */
+  private $params;
+
+  /**
+   * @var \Civi\API\SelectQuery
+   *  The SelectQuery instance wrapped by this class
+   */
+  private $query;
+
+  public function __construct($params) {
+    $this->params = $params;
+    $this->buildCustomQuery();
+  }
+
+  /**
+   * Build the custom query.
+   */
+  private function buildCustomQuery() {
+    $customQuery = CRM_Utils_SQL_Select::from(Comment::getTableName() . ' as a');
+
+    $this->addWhere($customQuery);
+    $this->filterReturnFields();
+    $this->query = new SelectQuery(Comment::class, $this->params, false);
+    $this->query->merge($customQuery);
+  }
+
+  /**
+   * Add the conditions to the query.
+   *
+   * This where it is ensured that only non soft-deleted comments are returned
+   * deleted contracts and with balance changes.
```

Comment:
I guess this line is a leftover from the comment on the class you based yourself on